vendor/hide-promoted-tweets.js

```diff
@@ -0,0 +1,23 @@
+(exports => {
+	'use strict';
+
+	exports.hidePromotedTweets = waitFor => {
+		const seekAndDestroy = () => waitFor('.vjrx_CgX').then(el => {
+			el.closest('div[class*="_222QxFjc"][role="row"]').style.display = 'none';
+		});
+
+		waitFor('._1nQuzuNK._3tixQkQf > ._3tixQkQf').then(tweetContainer => {
+			// hide any immediately seen ads
+			seekAndDestroy();
+
+			// watch tweetContainer to hide new ads that get added
+			new MutationObserver(() => {
+				seekAndDestroy();
+			}).observe(tweetContainer, {attributes: true, childList: true});
+		});
+	};
+
+	if (typeof module !== 'undefined' && module.exports) {
+		module.exports = window.hidePromotedTweets;
+	}
+})(window || module.exports);
```

Comment:
Just wondering why it's better? My solution looks a little cleaner to me but might not cover all use cases?